text/0000-app-namespacing-semantic-convention.md

```diff
@@ -0,0 +1,61 @@
+# Application namespacing semantic convention
+
+Adds a semantic convention for manual instrumentation that defines how to namespace custom attributes, starting with the `app` prefix.
+
+## Motivation
+
+Today, custom attributes from an app created through manual instrumentation have all kinds of names that are specific to the domain of that app. There are no conventions for how to organize this data, though.
+
+Data organization is critical to being able to effectively analyze observability data and avoid collisions in names. Without a well-defined convention for developers to follow, their own means of organization can differ from developer to developer, leading to a need to do things like coalesce field names using `.`/`-`/`_`/etc, not to mention having a real possibility to have name collisions. to separate clauses in an attribute name. Instead, a well-documented convention for how to organize data and in what format removes a "need to think" like code formatting conventions can do.
```

Comment:
This kind of high-level naming convention is already defined at https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-naming.md